File: 374_guess_number_higher_or_lower.java
/* The guess API is defined in the parent class GuessGame.
   @param num, your guess
   @return -1 if my number is lower, 1 if my number is higher, otherwise return 0
      int guess(int num); */

public class Solution extends GuessGame {
    public int guessNumber(int n) {
        int lower = 1, upper = n;
        while (lower <= upper) {
            int ans = lower + (upper - lower) / 2;
            int res = guess(ans);
            if (res == -1) {
                upper = ans - 1;
            } else if (res == 1) {
                lower = ans + 1;
            } else {
                return ans;
            }
        }
        return 1;
    }
}

File: python/374_guess_number_higher_or_lower.py
# The guess API is already defined for you.
# @param num, your guess
# @return -1 if my number is lower, 1 if my number is higher, otherwise return 0
# def guess(num: int) -> int:

class Solution:
    def guessNumber(self, n: int) -> int:
        low, high = 1, n
        while low < high:
            mid = low + (high - low) // 2
            if guess(mid) == 0:
                return mid
            elif guess(mid) == 1:
                low = mid + 1
            else:
                high = mid
        return low
